docs/html/reference/requirements-file-format.md

```diff
@@ -0,0 +1,144 @@
+# Requirements File Format
+
+Requirements files serve as a list of items to be installed by pip, when
+using {ref}`pip install`. Files that use this format are often called
+"pip requirements.txt files", since `requirements.txt` is usually what
+these files are named (although, that is not a requirement).
+
+```{note}
+This is NOT a format that other tools should try to consume.
+
+It depends on various internal details of how pip works (eg: pip's CLI
+options) and is not designed for interoperability.
+```
```

Comment:
Hmm... How do folks feel about this note in a dedicated page for the requirements file format?

Reply:
I feel like we should drop the first line, since it's not like we're going to be able to treat that as binding in any way; given the current state of affairs.

Reply:
I feel like it's valid for us to say, but maybe we word it differently?

> The requirements file format is closely tied to a number of internal details of pip (e.g., pip's command line options). The basic format is relatively stable and portable, but the full syntax, as described here, is only intended for consumption by pip, and other tools should take that into account before using it for their own purposes.

Also, do we want to note that (as it's technically an internal format) we reserve the right to change it when we release new versions of pip? (I'm mostly thinking about option names here, but in theory we could change other details as well).

Reply:
Agreed, I think it's implied, and TBH "people would have to change their requirements files" is one of the compatibility constraints that would *stop* us changing an option name. So maybe it's not worth it.

Thinking about it another way, is there even an issue here - does anyone even think that a requirements file is usable outside of pip? I suspect there are people who do, but they are actually thinking of some sort of limited format that's "requirements file without the pip-specific bits". Which of course may be subtly different in other ways. I don't know if we can usefully do anything to point out that "pip consumes requirement files as defined here, don't assume that whatever you think of as a requirements file is the same without checking"... 🤷

Reply:
I don't think anyone tries to parse and use the file like pip does.

Almost all uses I've seen are trying to read and spew out a modified version, define their own subset of the format, or generate a requirements file from some other format/data. I think the current wording covers them fairly well. They'll likely also call it `requirements.txt` files; so, yea, that overloading is part of why I have tried to restrict the format to disallow certain usecases in the past. :)